zcash_script/src/lib.rs

```diff
@@ -0,0 +1,462 @@
+                                        return Err(ScriptError::UnbalancedConditional);
+                                    }
+                                    todo!()
+                                }
+                                exec.push(value);
+                            }
+                            Opcode::OP_ELSE => {
+                                if exec.is_empty() {
+                                    return Err(ScriptError::UnbalancedConditional);
+                                }
+
+                                exec.last_mut().map(|last| *last = !*last);
+                            }
+                            Opcode::OP_ENDIF => {
+                                if exec.is_empty() {
+                                    return Err(ScriptError::UnbalancedConditional);
+                                }
+
+                                exec.pop();
+                            }
+                            Opcode::OP_VERIFY => {
+                                if stack.is_empty() {
+                                    return Err(ScriptError::InvalidStackOperation);
+                                }
+
+                                let value = stack.pop().unwrap();
+
+                                todo!()
+                            }
+                            Opcode::OP_RETURN => return Err(ScriptError::OpReturn),
+                            Opcode::OP_TOALTSTACK => {
+                                if stack.is_empty() {
+                                    return Err(ScriptError::InvalidStackOperation);
+                                }
+
+                                alt_stack.push(stack.pop().unwrap());
+                            }
+                            Opcode::OP_FROMALTSTACK => {
+                                if alt_stack.is_empty() {
+                                    return Err(ScriptError::InvalidStackOperation);
+                                }
+
+                                stack.push(alt_stack.pop().unwrap());
+                            }
+                            Opcode::OP_2DROP => {
+                                if stack.len() < 2 {
+                                    return Err(ScriptError::InvalidStackOperation);
+                                }
+
+                                stack.pop();
+                                stack.pop();
+                            }
+                            Opcode::OP_2DUP => {
+                                if stack.len() < 2 {
+                                    return Err(ScriptError::InvalidStackOperation);
+                                }
+
+                                let b = stack.pop().unwrap();
+                                let a = stack.pop().unwrap();
+                                stack.push(a.clone());
+                                stack.push(b.clone());
+                                stack.push(a);
+                                stack.push(b);
```

Comment:
All of these implementations could be made more concise with some helper functions ({push/pop}{1,2,3,4}).